pkg/eventfilter/filter_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventfilter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/stretchr/testify/require"
+)
+
+type mockFilter FilterResult
+
+func (p mockFilter) Filter(ctx context.Context, event cloudevents.Event) FilterResult {
+	return FilterResult(p)
+}
+
+func TestFilters(t *testing.T) {
+	tests := []struct {
+		have []FilterResult
+		want FilterResult
```

Comment:
Shouldn't this be `NoFilter` instead?

Reply:
This is the type of the wanted filter, not the wanted filter itself. I think it's right this way.